### DISCUSSION TITLE
Composition of multiple projects from static files instead of URLs

### DISCUSSION THREAD

**Discussion by alreadyhaveanaccount4** [storybookjs/storybook#20773] [Help]
Hello there,

I have 3 projects `P1` `P2a` `P2b`. I want to host them all in one.

In `P1` i have a `main.js` inside `.storybook` using `refs` like this: 

```
module.exports={
  // your Storybook configuration
  refs: {
   'P2a': {
     title: "Project 2a",
     url: "https://myurl.com/a",
   }
   'P2b': {
     title: "Project 2b",
     url: "https://myurl.com/b",
   }
  }
}
```

But this requires *all three projects* to be up and running at the same time. 
I have published all three of them using `yarn build-storybook`, so I have three `storybook-static` folders like this:
```
P1/storybook-static/
P2a/storybook-static/
P2b/storybook-static/
```

Is there an alternative to use a **path** to a static storybook **instead of** hosting them live on a **URL** ?
Something like this:

```
refs: {
   'P2a': {
     title: "Project 2a",
     path: "projects/P2a/storybook-static",
   }
....
```

Using all of this in Angular. Thanks in advance.

**Comment by mattfelten** [comment]
I just had this same question. Running Storybook locally can work, you just have to spin up each instance as separate servers. We're serving our production Storybook just as static files though, and we lose out on composition that way. It would be great to be able to link to static files.

Ideally we should be able to set a URL _and_ static files as a fallback, so if the URL doesn't exist it can use the static files if they exist.

**Reply by airtonix** [reply]
you're overthinking it.

**Comment by airtonix** [comment]
Would this work? 

```
yarn build-all-the-things
yarn deploy-all-the-things
# prints deployed to vercel at https://magicalurl.somethi.ng/pr/234
```

```
P1 -> https://magicalurl.somethi.ng/pr/234/storybook-p1/
P2a -> https://magicalurl.somethi.ng/pr/234/storybook-p2b/
P2b -> https://magicalurl.somethi.ng/pr/234/storybook-p2a/
```


config could be : 

```
module.exports={
  // your Storybook configuration
  refs: {
   'P2a': {
     title: "Project 2a",
     url: "../storybook-p2a",
   }
   'P2b': {
     title: "Project 2b",
     url: "../storybook-p2b",
   }
  }
}
```

**Reply by lems3** [reply]
This is exactly what OP try to avoid. As I do. I'd rather not have to publicly host every Storybook I want to compose in other projects. We want to install the library as a dependency in our projects, then spin one Storybook per project we have.

The goal is to have one central Design Library, and let each project define how they will use them in their complex components.
